### ISSUE TITLE
Realm issue after upgrading the react-native@0.71.11 and realm@11.10.1

### ISSUE THREAD

**Issue by Atulpal9045** [realm/realm-js#5973]
### How frequently does the bug occur?

Always

### Description

Hey folks, 

I upgraded my react-native app from react-native@0.62.2 to react-native@0.71.11 with realm@11.10.1 and it's working fine in the debug mode but when I'm trying to run it to the release mode it's crashing after few seconds. 

### Stacktrace & log output

```shell
9762  9810 E unknown:NativeModuleInitError: Failed to create NativeModule "Realm"
9762  9810 E unknown:NativeModuleInitError: java.lang.UnsatisfiedLinkError: couldn't find DSO to load: librealm.so caused by: io.realm.react.util.SSLHelper result: 0
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader.g(Unknown Source:341)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader.t(Unknown Source:124)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader.s(Unknown Source:2)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader.q(Unknown Source:42)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader.p(Unknown Source:1)
9762  9810 E unknown:NativeModuleInitError:         at io.realm.react.RealmReactModule.<clinit>(Unknown Source:2)
9762  9810 E unknown:NativeModuleInitError:         at io.realm.react.a.e(Unknown Source:8)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.z$b.a(Unknown Source:6)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.z$b.get(Unknown Source:0)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.bridge.ModuleHolder.create(Unknown Source:67)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.bridge.ModuleHolder.getModule(Unknown Source:20)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.bridge.JavaModuleWrapper.findMethods(Unknown Source:14)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.bridge.JavaModuleWrapper.getMethodDescriptors(Unknown Source:8)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.jni.NativeRunnable.run(Native Method)
9762  9810 E unknown:NativeModuleInitError:         at android.os.Handler.handleCallback(Handler.java:942)
9762  9810 E unknown:NativeModuleInitError:         at android.os.Handler.dispatchMessage(Handler.java:99)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0)
9762  9810 E unknown:NativeModuleInitError:         at android.os.Looper.loopOnce(Looper.java:201)
9762  9810 E unknown:NativeModuleInitError:         at android.os.Looper.loop(Looper.java:288)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(Unknown Source:37)
9762  9810 E unknown:NativeModuleInitError:         at java.lang.Thread.run(Thread.java:1012)
9762  9810 E unknown:NativeModuleInitError: Caused by: java.lang.ClassNotFoundException: io.realm.react.util.SSLHelper
9762  9810 E unknown:NativeModuleInitError:         at java.lang.Runtime.nativeLoad(Native Method)
9762  9810 E unknown:NativeModuleInitError:         at java.lang.Runtime.nativeLoad(Runtime.java:1121)
9762  9810 E unknown:NativeModuleInitError:         at java.lang.Runtime.load0(Runtime.java:927)
9762  9810 E unknown:NativeModuleInitError:         at java.lang.System.load(System.java:1625)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader$a.a(Unknown Source:201)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.d.h(Unknown Source:176)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.d.a(Unknown Source:2)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.b.a(Unknown Source:2)
9762  9810 E unknown:NativeModuleInitError:         at com.facebook.soloader.SoLoader.g(Unknown Source:67)
9762  9810 E unknown:NativeModuleInitError:         ... 20 more
9762  9810 E AndroidRuntime: FATAL EXCEPTION: mqt_js
9762  9810 E AndroidRuntime: Process: com.myapp, PID: 9762
9762  9810 E AndroidRuntime: java.lang.RuntimeException: com.facebook.react.devsupport.JSException: Exception in HostObject::get(propName:Realm): java.lang.UnsatisfiedLinkError: couldn't find DSO to load: librealm.so caused by: io.realm.react.util.SSLHelper result: 0
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.DefaultJSExceptionHandler.handleException(Unknown Source:9)
9762  9810 E AndroidRuntime:        at com.facebook.react.devsupport.c.handleException(Unknown Source:2)
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.CatalystInstanceImpl.onNativeException(Unknown Source:2)
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.CatalystInstanceImpl.access$1100(Unknown Source:0)
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.CatalystInstanceImpl$NativeExceptionHandler.handleException(Unknown Source:2)
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:7)
9762  9810 E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)        
9762  9810 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(Unknown Source:37)
9762  9810 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:1012)
9762  9810 E AndroidRuntime: Caused by: com.facebook.react.devsupport.JSException: Exception in HostObject::get(propName:Realm): java.lang.UnsatisfiedLinkError: couldn't find DSO to load: librealm.so caused by: io.realm.react.util.SSLHelper result: 0
9762  9810 E AndroidRuntime:        at com.facebook.jni.NativeRunnable.run(Native Method) 
9762  9810 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:942)
9762  9810 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
9762  9810 E AndroidRuntime:        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(Unknown Source:0)
9762  9810 E AndroidRuntime:        ... 4 more
9762  9810 E AndroidRuntime: Caused by: com.facebook.jni.CppException: Exception in HostObject::get(propName:Realm): java.lang.UnsatisfiedLinkError: couldn't find DSO to load: librealm.so caused by: io.realm.react.util.SSLHelper result: 0
9762  9810 E AndroidRuntime:
9762  9810 E AndroidRuntime: index.android.bundle:735:431
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: index.android.bundle:734:41
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: index.android.bundle:732:79
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: index.android.bundle:10:735
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: index.android.bundle:9:111
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: index.android.bundle:8:111
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: index.android.bundle:6:45
9762  9810 E AndroidRuntime: _@index.android.bundle:2:1705
9762  9810 E AndroidRuntime: d@index.android.bundle:2:1083
9762  9810 E AndroidRuntime: global code@index.android.bundle:991:4
```

### Can you reproduce the bug?

Always

### Reproduction Steps

`npm run android -- --mode release`

### Version

11.10.1

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

Windows 11 Pro 

### Build environment

Which debugger for React Native: ..
react-native: 0.71.11
realm: 11.10.1
node:18.16.1 
npm:8.19.4

these are the other dependencies - 

  "dependencies": {
    "@react-native-async-storage/async-storage": "^1.17.11",
    "@react-native-community/cli": "^11.3.3",
    "@react-native-firebase/analytics": "^18.1.0",
    "@react-native-firebase/app": "^18.1.0",
    "@react-native-firebase/crashlytics": "^18.1.0",
    "@react-native-firebase/messaging": "^18.1.0",
    "@react-native-firebase/remote-config": "^18.1.0",
    "@react-native/metro-config": "^0.72.6",
    "hermes-engine": "^0.11.0",
    "jwt-decode": "^3.1.2",
    "lodash": "^4.17.21",
    "lottie-ios": "^3.5.0",
    "lottie-react-native": "^5.1.6",
    "moment": "^2.29.4",
    "patch-package": "^7.0.0",
    "react": "18.2.0",
    "react-native": "^0.71.11",
    "react-native-android-location-enabler": "^1.2.2",
    "react-native-confirmation-code-input": "^1.0.4",
    "react-native-device-info": "^10.7.0",
    "react-native-easy-grid": "^0.2.2",
    "react-native-geolocation-service": "^5.3.1",
    "react-native-image-picker": "^5.6.0",
    "react-native-keychain": "^8.1.1",
    "react-native-localization": "^2.3.2",
    "react-native-modal": "^13.0.1",
    "react-native-navigation": "^7.32.1",
    "react-native-sms-retriever": "^1.1.1",
    "react-native-uuid-generator": "^6.1.1",
    "react-native-webview": "^13.2.2",
    "realm": "^11.10.1"
  },

**Comment by kneth** [comment]
@Atulpal9045 You might have to disable the new architecture (it has been fixed in v12.0.0-rc.0)

**Comment by dulajappspotr** [comment]
I faced the same issue. Any solution for this issue?
